Add Modbus Connection integration documentation (#46632)
home-assistant/home-assistant.github.io: source/_integrations/modbus_connection.markdown

Modbus is a serial communications protocol widely used in industrial and building automation to talk to devices such as power meters, PLCs, HVAC controllers, and inverters. A physical Modbus link usually accepts only one client at a time, so it cannot be opened independently by several integrations at once.

This integration solves that by owning the connection in one place. You set up each connection once, and then point Modbus device integrations at it. This is the UI-based way to use Modbus in Home Assistant. If you prefer to register Modbus entities manually in YAML, use the [Manual Modbus](/integrations/modbus/) {% term integration %} instead.

This {% term integration %} does not create any entities of its own. It only provides the shared connection that device integrations consume.

## Prerequisites

You need one of the following ways to reach your Modbus network:

- For a **network** connection: a Modbus TCP gateway or a device that speaks Modbus TCP (or RTU-over-TCP), reachable by hostname or IP address on your network.

> - For a **serial** connection: a serial (RTU) device connected to the machine running Home Assistant, for example a USB-to-RS485 adapter appearing as `/dev/ttyUSB0`. A network serial proxy is also supported and appears in the serial device list.

Reviewer: ```suggestion
- For a *network* connection: a Modbus TCP gateway or a device that speaks Modbus TCP (or RTU-over-TCP), reachable by hostname or IP address on your network.
- For a *serial* connection: a serial (RTU) device connected to the machine running Home Assistant, for example a USB-to-RS485 adapter appearing as `/dev/ttyUSB0`. A network serial proxy is also supported and appears in the serial device list.
```

{% include integrations/config_flow.md %}

When you add the integration, you first choose how the Modbus network is connected:

- **Network**: for Modbus TCP and RTU-over-TCP connections.

> - **Serial**: for Modbus RTU connections over a serial port, including network serial proxies.

Reviewer: ```suggestion 
- **Modbus TCP**: for Modbus TCP and RTU-over-TCP connections.
- **Serial (including serial proxies and networked connections)**: for Modbus RTU connections over a serial port.
```

> ### Network

Reviewer: ```suggestion
### Modbus TCP
```
